### DISCUSSION TITLE
two auto profiles overlap

### DISCUSSION THREAD

**Discussion by marat569** [mpv-player/mpv#14062] [Q&A] [answered]
mpv version: mpv-x86_64-20240505-git-c4b6d0d
windows 11
nvidia gpu

Sorry for the issue, but my question is simple; and you guys are the best place to ask

In my MPV.conf I have two auto profiles:

```
[hdr]
profile-desc="HDR"
profile-cond=p.video_params.gamma == "pq" or p.video_params.gamma == "hlg"
target-peak=1000
tone-mapping=bt.2446a
contrast=0 # Reset contrast value

#Profile for loading AVIF HDR images [will add JXR when support is added]
[hdrImages]
profile-cond=get('filename', ''):match('%.avif$') ~= nil
contrast=0 #reset contrast
hwdec=no #avoids hardware decoding errors
deband=no
```

the goal is for [hdr] to load when playing hdr video, and [hdrImages] to load only when opening avif (and jxr) HDR images in the future.

The current behavior is both [hdr] and [hdrImages] get loaded when opening a HDR .AVIF image, causing target-peak and tone-mapping to get inherited from [hdr] -- something I don't want -- I'd like [hdrImages] to have essentially no "post-processing" -- just display the image

I know I can add a bunch of extension checks to the [HDR] profile, similar to how [hdrImages] checks if a file is a .avif file -- but I'm hoping to avoid that. 

I was wondering if there is a flag I can add to [hdrImages] so it ignores mpv.conf or the [hdr] auto profile

Thanks guys!

**Comment by marat569** [comment]
I think I found a hacky workaround, basically for [HDR] I check if container_fps is >1

```
[hdr]
profile-desc="HDR"
profile-cond=(p.video_params.gamma == "pq" or p.video_params.gamma == "hlg") and container_fps > 1
#do stuff
```

this way [hdr] loads only when playing back hdr video, and [hdrImages] loads only when opening hdr images 

I'm keeping this open incase there is a more "elegant"/"proper" way of handling things

**Comment by guidocella** [comment, marked as answer]
`p['current-tracks/video/image'] == false`

**Comment by marat569** [comment]
> `p['current-tracks/video/image'] == false`

thanks a lot man, I spent a while googling, asking chatgpt, etc -- this is probably the best solution. I really appreciate it.

Here are the auto-profiles I ended up making.

```
#HDR Profile that auto activates when Native HDR media is played (ty astralstorm)
[hdr]
profile-desc="HDR"
profile-cond=(p.video_params.gamma == "pq" or p.video_params.gamma == "hlg") and p['current-tracks/video/image'] == false
target-peak=1000
tone-mapping=bt.2446a
contrast=0 # Reset contrast value

#Profile for loading AVIF HDR images [will add JXR when support is added]
[hdrImages]
profile-cond=get('filename', ''):match('%.avif$') ~= nil
contrast=0 #reset contrast
hwdec=no #avoids hardware decoding errors
deband=no
```

**Comment by guidocella** [comment]
You may want to use `p['current-tracks/video/codec'] == 'av1'` and `p['current-tracks/video/image']` to not rely on the filename being correct. Your condition also won't work on urls with a query string after `avif`.

**Reply by marat569** [reply]
I ended up doing this, so it loads all HDR images -- so i don't have to edit anything when we get JXR images:

```
#Profile for loading hdr images
[hdrImages]
#profile-cond=get('filename', ''):match('%.avif$') ~= nil
profile-cond=(p.video_params.gamma == "pq" or p.video_params.gamma == "hlg") and p['current-tracks/video/image'] == true
contrast=0 #reset contrast
hwdec=no #avoids hardware decoding errors
deband=no
```

I just had to add this to the bottom of my conf, since the check is too late to apply hwdec=no properly:
```
[extension.avif]
hwdec=no #hwdec dun work
```

**Comment by guidocella** [comment]
Should we not try to hardware decode images? I can hardware decode at least at least baseline jpgs with vaapi (but not progressive ones) but it doesn't make any visible difference.

**Reply by marat569** [reply]
I'm sure some work, but when it comes to HDR .AVIF iamges, I get a ton of spam on my 4090 that goes "unable to decode image" and then it gives up

**Reply by Samillion** [reply]
Sorry for reviving an old discussion, was searching for something related to this and was going to ask the same thing.

Do image viewers typically use hardware decoding? As far as I know it's always software, I could be wrong though.

On top of decoding errors with some images, high res ones with software decode just feels more snappy, if that makes sense.

Using a conditional profile for images with `hwdec=no` is definitely not consistent, for example in a library with mixed types (images and videos)

**Reply by guidocella** [reply]
They don't use hardware decoding (at least on desktop).

Like already noted conditional profiles are also evaluated too late to prevent decoding errors, you have to use `[extension.avif]`. (For older codecs that are only used for images they are simply not in the default `--hwdec-codecs`). So we may need to add an option to disable hardware decoding on images.

**Reply by Samillion** [reply]
> So we may need to add an option to disable hardware decoding on images.

That would be awesome.
